examples/hpc-enterprise-slurm.yaml

```diff
@@ -46,6 +46,36 @@ deployment_groups:
   - id: network1
     source: modules/network/pre-existing-vpc
 
+  - id: controller_sa
```

Comment:
Don't you need to create these in a previous group? Or is terraform trustworthy here in terms of lifecycle?

Reply:
There are 2 potential problems I see:

1. Terraform is satisfied once the SA is created (a global resource) but there is a subsequent latency in propagation of IAM permissions across regions. I cannot find the link but we addressed some of this concern in #1073
2. There are conditions in which the SA can be destroyed and re-created with the same e-mail address but a new unique ID (a large number). In this case, a VM in a subsequent group may be satisfied in the Terraform sense (nothing to do on apply) but actually have a non-existent SA attached to it.

2 is not a problem within a single group (because a single Terraform apply sees the connection).  It can be resolved by:

- rebooting the VM 
- recreating the VM

A thing we could do is also pass along the unique_id and trigger recreation based upon the unique ID. I'd want @mr0re1 to think about adding the unique_id attribute to this object.

Reply:
I am worried about permissions to access services.. For instance, a VM being created without GCS scope. Wouldn't that fail *before* getting to our startup scripts?

Reply:
What is possible is that a VM will be created with a valid service account whose IAM roles have not yet propagated to the region where the VM is being created. Scopes are a property of the VM, not the service account, so I don't think you can accidentally not have the scopes. Additionally, the underlying message of this PR is that "cloud-platform" is the right scope for custom service accounts in the first place.

The options here are:

1. Enforce a delay upon creation of a new service account hat achieves X 9s of reliability for attaching it a to GCP resource ([SLA guidance](https://cloud.google.com/iam/docs/access-change-propagation))
2. Make solutions that are resilient against transient failures

(1) makes initial deployment slower for no good reason in most cases. (2) is the right thing to do for a long list of reasons. Thus, I would focus on (2) as in #1073